controller/internal/enforcer/envoyauthorizer/envoyproxy/auth_server.go

```diff
@@ -0,0 +1,396 @@
+	// TODO: needs to be removed before we merge: this exposes secret data, and must never be in real logs - even in the debug case
+	zap.L().Debug("ext_authz ingress: checkRequest", zap.String("puID", s.puID), zap.String("checkRequest", checkRequest.String()))
+
+	// get the PU context
+	// TODO: check with marcus, not sure if we want a LOCK here, as this can be accessed now by the
+	// 1. Envoy auth server.
+	// 2. by enforcer policy update.
+	// pctxRaw, err := s.puContexts.Get(s.puID)
+	// if err != nil {
+	// 	zap.L().Error("ext_authz ingress: failed to get PU context", zap.String("puID", s.puID), zap.Error(err))
+	// 	return createDeniedCheckResponse(rpc.INTERNAL, envoy_type.StatusCode_InternalServerError, "failed to get PU context"), nil
+	// }
+	// pctx, ok := pctxRaw.(*pucontext.PUContext)
+	// if !ok {
+	// 	zap.L().Error("ext_authz ingress: PU context has the wrong type", zap.String("puID", s.puID), zap.String("puContextType", fmt.Sprintf("%T", pctxRaw)))
+	// 	return createDeniedCheckResponse(rpc.INTERNAL, envoy_type.StatusCode_InternalServerError, "PU context has the wrong type"), nil
+	// }
+	// now extract the attributes and call the API auth to decode and check all the claims in request.
+	var sourceIP, destIP, aporetoAuth, aporetoKey string
+	var source, dest *ext_auth.AttributeContext_Peer
+	var httpReq *ext_auth.AttributeContext_HttpRequest
+	var destPort, srcPort int
+	var urlStr, method string
+	attrs := checkRequest.GetAttributes()
+	if attrs != nil {
+		source = attrs.GetSource()
+		dest = attrs.GetDestination()
+
+		if source != nil {
+			if addr := source.GetAddress(); addr != nil {
+				if sockAddr := addr.GetSocketAddress(); sockAddr != nil {
+					sourceIP = sockAddr.GetAddress()
+					srcPort = int(sockAddr.GetPortValue())
+				}
+			}
+		}
+		if dest != nil {
+			if destAddr := dest.GetAddress(); destAddr != nil {
+				if destSockAddr := destAddr.GetSocketAddress(); destSockAddr != nil {
+					destIP = destSockAddr.GetAddress()
+					destPort = int(destSockAddr.GetPortValue())
+				}
+			}
+		}
+
+		if request := attrs.GetRequest(); request != nil {
+			httpReq = request.GetHttp()
+			if httpReq != nil {
+				httpReqHeaders := httpReq.GetHeaders()
+				aporetoAuth, _ = httpReqHeaders[aporetoAuthHeader]
+				aporetoKey, _ = httpReqHeaders[aporetoKeyHeader]
+				fmt.Println("httpReqheader is :", httpReqHeaders, "aporeto key is:", aporetoKey)
+				urlStr = httpReq.GetPath()
+				method = httpReq.GetMethod()
+
+			}
+		}
+	}
+	fmt.Println("in auth check, source addr: ", sourceIP, "source, dest: ", source.GetAddress().GetSocketAddress().GetAddress(), dest.GetAddress().GetSocketAddress().GetAddress())
+	fmt.Println("dset port: ", destPort, "src port: ", srcPort)
+	requestCookie := &http.Cookie{Name: aporetoAuthHeader, Value: aporetoAuth} // nolint errcheck
+	hdr := make(http.Header)
+	//URL := url.URL{}
+	//hdr.Add()
+	hdr.Add(aporetoAuthHeader, aporetoAuth) //string(p.secrets.TransmittedKey()))
```

Comment:
TODO: Add the auth header and cookie for user creds.

Reply:
... jog my memory, what's this one about again?

Reply:
not sure as I already adding the auth header and cookie.